src/vos/tests/vos_size.c

```diff
@@ -0,0 +1,180 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/* This generates a summary of struct sizes to be used by vos_estimate.py
+ * to generate metadata overhead estimates
+ */
+#define D_LOGFAC	DD_FAC(tests)
+
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <getopt.h>
+#include <daos/debug.h>
+#include <daos_srv/vos.h>
+
+#define FOREACH_TYPE(ACTION)					\
+	ACTION(container, VOS_TC_CONTAINER, 0)			\
+	ACTION(object, VOS_TC_OBJECT, 0)			\
+	ACTION(dkey, VOS_TC_DKEY, 0)				\
+	ACTION(akey, VOS_TC_AKEY, 0)				\
+	ACTION(integer_dkey, VOS_TC_DKEY, DAOS_OF_DKEY_UINT64)	\
+	ACTION(integer_akey, VOS_TC_AKEY, DAOS_OF_AKEY_UINT64)	\
+	ACTION(single_value, VOS_TC_SV, 0)			\
+	ACTION(array, VOS_TC_ARRAY, 0)
+
+#define DECLARE_TYPE(name, type, feats)	\
+	struct daos_tree_overhead	name;
+
+#define CHECK_CALL(name, type, feats)					\
```

Comment:
(style)  Macros with flow control statements should be avoided

(style)  do {} while (0) macros should not be semicolon terminated